### PR TITLE
Data set contributors

### PR DESCRIPTION
**Pull request by jstone-dev**
Creators of experiments and score sets may now assign them lists of contributors. Contributors must be looked up by ORCID ID, and the API returns the ORCID user's name. If a contributor has a MaveDB user account, that user is granted the same permissions on the experiment or score set as its creator.

- User interfaces for assigning contributors to an experiment or score set
- Experiment and score set views now display lists of contributors.
- Pinia and rest-client-vue are used here to manage ORCID ID lookups. We'll likely want to replace other uses of Vuex and useItems with these libraries.

This PR depends on the corresponding contributors PR on mavedb-api (https://github.com/VariantEffect/mavedb-api/pull/259).

### CODE REVIEW

File: src/components/screens/ScoreSetCreator.vue
Comment (bencap) [inline src/components/screens/ScoreSetCreator.vue:335]: Contributors help text

File: src/components/screens/ScoreSetCreator.vue
Comment (bencap) [inline src/components/screens/ScoreSetCreator.vue:342]: We should consider adding the contributors to the list of things auto-populated by experiment. Alan noted that most of the time are the same between the two items so it'd be nice for that to be the default. 

We'd just need to add something like
`this.contributors = event.value.contributors` in `populateExperimentMetadata`.

File: src/components/screens/ScoreSetEditor.vue
Comment (bencap) [inline src/components/screens/ScoreSetEditor.vue:821]:
```suggestion
```

File: src/components/screens/ScoreSetCreator.vue
Comment (bencap) [inline src/components/screens/ScoreSetCreator.vue:1065]:
```suggestion
```

File: src/components/screens/ScoreSetCreator.vue
Comment (bencap) [inline src/components/screens/ScoreSetCreator.vue:1255]:
```suggestion
```

**Review by bencap — COMMENTED** [review]
Thanks Jeremy, looks good overall. Left some minor comments-- I can approve once we take this off of draft, you may still have some local changes.

File: src/components/screens/ExperimentEditor.vue
Comment (jstone-dev) [reply src/components/screens/ExperimentEditor.vue:212]: Same here. It doesn't seem to have to do with the floating label, which was my first guess, or with the use of slots. I don't mind the behavior in this case, especially with the help text to the left; but I'd like to know what's causing it.

File: src/components/screens/ExperimentEditor.vue
Comment (jstone-dev) [reply src/components/screens/ExperimentEditor.vue]: Done

File: src/components/screens/ExperimentEditor.vue
Comment (jstone-dev) [reply src/components/screens/ExperimentEditor.vue]: Yep, thanks.

File: src/components/screens/ScoreSetCreator.vue
Comment (jstone-dev) [reply src/components/screens/ScoreSetCreator.vue:338]: It's populated now from either the experiment or the superseded score set.

File: src/lib/orcid.ts
Comment (bencap) [inline src/lib/orcid.ts:46]: ORCID IDs may also end with an `X`.

```suggestion
export const ORCID_ID_REGEX = /^\d{4}-\d{4}-\d{4}-(\d{4}|\d{3}X)$/
```

**Review by bencap — COMMENTED** [review]
Looks nice, thanks for the Chips change. Once the ORCID Regex is changed we can merge this in.

In addition to the ORCID thing: The warning keyword for toast notifications is now `warn` not `warning` (unrelated but it seems like a lot of libraries are deprecating `warning`-- the python default logger is doing so as well). We lose message styling when we pass `warning` rather than `warn`.

File: src/lib/orcid.ts
Comment (jstone-dev) [reply src/lib/orcid.ts]: Fixed -- thanks for pointing this out.

**Review by bencap — APPROVED** [review]

**Pull request merged**
